src/TSTransformer/util/types.ts

```diff
@@ -245,3 +245,16 @@ export function getFirstDefinedSymbol(state: TransformState, type: ts.Type) {
 export function getTypeArguments(state: TransformState, type: ts.Type) {
 	return state.typeChecker.getTypeArguments(type as ts.TypeReference) ?? [];
 }
+
+export function isPossiblyMacroIterationType(state: TransformState, type: ts.Type) {
+	return isPossiblyType(
```

Comment:
This feels like a very duplicated list (with the resulting "remove one, forget the other" pitfall). This might be suitable for a util, where you can pass in a list of `typecheck > callback` and the util takes care of checking there's no type mixing?